packages/flutter_riverpod/test/framework_test.dart

```diff
@@ -802,6 +812,20 @@ void main() {
   });
 }
 
+final multipleFutureProvider = FutureProvider<String>((ref) async => 'foo');
+
+class MultipleProviderScopeNotifier extends AutoDisposeNotifier<String> {
```

Comment:
Do we care about it being a Notifier?

If not, use a plain Provider. Then move the provider definition inside the test

Reply:
Yes this does seem to be related to a notifier provider refreshing another provider within its build method.

Reply:
Wait, this shouldn't even be legal to begin with.

It's considered not allowed to call `ref.refresh` inside a provider's init method.

Reply:
Updated to use a different mechanism